### ISSUE TITLE
infer-onnx / infer-trt の推論ループに DataLoader を導入してデータ読み込みを高速化

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#191]
## 概要

`infer-onnx` と `infer-trt` の推論ループでは `PochiImageDataset` をインデックスアクセス (`dataset[i]`) で逐次読み込みしており, DataLoader を使っていない. そのため `num_workers` によるバックグラウンドプリフェッチが効かず, 画像の読み込み・transform がメインスレッドで毎回実行される.

推論自体は高速だが, 画像の読み込み・transform がボトルネックとなり全処理時間を押し上げている.

### 実測値 (resnet50, 512x512, 38枚, リサイズなし)

| バックエンド | 純粋推論 | 全処理 | オーバーヘッド |
|---|---|---|---|
| ONNX (GPU) | 3.38 ms/image | 7.36 ms/image | 3.98 ms/image |
| TensorRT INT8 | 0.66 ms/image | 4.09 ms/image | 3.43 ms/image |

リサイズなしの transform (ToTensor のみ) でもオーバーヘッドの大部分は PIL による画像読み込みが占めており, DataLoader の `num_workers` で並列化することで改善が見込まれる.

## 問題の詳細

**現在のコード: infer_onnx.py (手動バッチ化)**

```python
# pochitrain/cli/infer_onnx.py:173-185
for batch_idx in range(num_batches):
    start_idx = batch_idx * batch_size
    end_idx = min(start_idx + batch_size, len(dataset))

    batch_images = []
    batch_labels = []
    for i in range(start_idx, end_idx):
        image, label = dataset[i]  # メインスレッドで逐次読み込み
        batch_images.append(image.numpy())
        batch_labels.append(label)

    images_np = np.stack(batch_images).astype(np.float32)
```

**現在のコード: infer_trt.py (1件ずつ読み込み)**

```python
# pochitrain/cli/infer_trt.py:183-186
for i in range(len(dataset)):
    image, label = dataset[i]  # メインスレッドで逐次読み込み
    image_np = image.numpy()[np.newaxis, ...].astype(np.float32)
```

**PyTorch 推論 (pochi infer) では DataLoader を使用済み:**

```python
# pochitrain/cli/pochi.py:411-415
val_loader = DataLoader(
    ...
    num_workers=pochi_config.num_workers,
)
```

## 修正方針

- `infer-onnx` / `infer-trt` の推論ループに `DataLoader` を導入
- `--num-workers` CLI オプションを追加 (デフォルト: 0)
- `pochi infer` と同様に `num_workers` で画像読み込みの並列化を有効化
- 推論時間の計測ロジック (ウォームアップ, CUDA Event) は維持

**注意**: 1枚推論 (バッチサイズ1, 画像1枚) の場合は `num_workers` による効果の期待は低い. 1枚推論の高速化は #192 で対応する.

## Labels

`enhancement`

## Branch

`feature/inference-dataloader`
